src/database/sql_database.cc

```diff
@@ -1604,9 +1603,8 @@ std::unique_ptr<Database::ChangedContainers> SQLDatabase::_purgeEmptyContainers(
     log_debug("start upnp: {}; ui: {}",
         join(maybeEmpty->upnp, ',').c_str(),
         join(maybeEmpty->ui, ',').c_str());
-    auto changedContainers = std::make_unique<ChangedContainers>();
     if (maybeEmpty->upnp.empty() && maybeEmpty->ui.empty())
-        return changedContainers;
+        return nullptr;
```

Comment:
These should be equivalent. The only place that uses this has a nullptr check: https://github.com/gerbera/gerbera/blob/master/src/content/content_manager.cc#L804